compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1050,9 +1050,9 @@ impl<'hir> LoweringContext<'_, 'hir> {
         asyncness: Async,
         body: Option<&Block>,
     ) -> hir::BodyId {
-        let closure_id = match asyncness {
-            Async::Yes { closure_id, .. } => closure_id,
-            Async::No => return self.lower_fn_body_block(span, decl, body),
+        let (closure_id, body) = match (asyncness, body) {
```

Comment:
This means that when a (non-trait) fn is lacking a body, we lower it to `<ExprKind::Err>` instead of like... ```async { <ExprKind::Err> }```